src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -254,19 +254,18 @@ object InstanceOpFactoryImpl {
     hostPorts: Seq[Option[Int]],
     instanceId: Instance.Id)(implicit clock: Clock): Instance = {
 
-    val hostPortsByContainerName: Map[String, Seq[Int]] = {
-      val portsByEndpointName = pod.hostPortsByEndpointName(hostPorts)
-      assume(
-        portsByEndpointName.size == hostPorts.size,
-        s"expected that number of available ports ${hostPorts.size}" +
-          s" would equal the number of requested host ports $portsByEndpointName")
-
-      pod.containers.map{ container =>
-        val containerPorts = container.endpoints.flatMap { ep => portsByEndpointName(ep.name) }
-        container.name -> containerPorts
-      }.toMap.withDefaultValue(Seq.empty[Int])
+    val requestedPortsPerCT: Seq[(String, Option[Int])] = pod.containers.flatMap { ct =>
```

Comment:
IMHO the suffix `ByContainerName` is much more explicit than `PerCT`, especially since the key is a String.

maybe `requestedPortsByContainerName`